### DISCUSSION TITLE
how to call usePrepareContractWrite conditionally

### DISCUSSION THREAD

**Discussion by Ding-Fan** [wevm/wagmi#804] [Q&A] [answered]
some function in the contract need correct arguments(like `foo` , `bar` in below) to do the gas estimate.

```ts
  const { config, error } = usePrepareContractWrite({
    addressOrName: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS as string,
    contractInterface: abi.abi,
    functionName: 'make',
    args: [foo, bar],
    overrides: {
      from: address,
      value: ethers.utils.parseEther('0.05'),
      // gasLimit: 1000000
    },
  })
```

When the `foo` and `bar` are `undefined`, this hook will cause an error  
`Error: cannot estimate gas; transaction may fail or may require manual gas limit`

Currently I'm trying to do this

```ts
// pseudo code
//
theFlag = useMemo({
  // set foo and bar as dependencies
  // and toggle theFlag 
})
...
usePrepareContractWrite({
...
enabled: theFlag
})
```

Is this normal?

**Comment by Ding-Fan** [comment]
related discussion https://github.com/wagmi-dev/wagmi/discussions/560

**Comment by tmm** [comment, marked as answer]
Yup! Using `enabled` is the way to do it.
